File: 3790-fruits-into-baskets-ii/3790-fruits-into-baskets-ii.py
class Solution:
    def numOfUnplacedFruits(self, fruits: List[int], baskets: List[int]) -> int:
        unplaced_fruits_count = 0
        placed = set()

        for fruit in fruits:
            for idx in range(len(baskets)):
                if idx not in placed and fruit <= baskets[idx]:
                    placed.add(idx)
                    break
        
            # print(placed)

        return len(baskets) - len(placed)

File: 3790-fruits-into-baskets-ii/3790-fruits-into-baskets-ii.java
class Solution {
    public int numOfUnplacedFruits(int[] fruits, int[] baskets) {

        HashSet<Integer> placed = new HashSet<>();

        for(int fruit : fruits){
            for(int idx = 0; idx < baskets.length; idx++){
                if(!placed.contains(idx) && fruit <= baskets[idx]){
                    placed.add(idx);
                    break;
                }
            }
        }

        int unplacedFruits = baskets.length - placed.size();
        return unplacedFruits;
    }
}